components/TickerSearch/TickerSearch.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react"
+import { catchError, debounceTime, map, switchMap } from "rxjs/operators"
+import styles from "./TickerSearch.module.scss"
+import { createSignal, mergeWithKey } from "@react-rxjs/utils"
+import { bind } from "@react-rxjs/core"
+import { ajax } from "rxjs/ajax"
+import { ITickerSearchResult } from "../../models/TickerSearchResult"
+import { of } from "rxjs"
+
+const [searchInputs$, onNextSearchInput] = createSignal<string>()
+
+const [tickerSelections$, onTickerSelection] = createSignal<string>()
+
+interface TickerOption {
+  value: string
+  label: string
+}
+
+const searchResults$ = searchInputs$.pipe(
+  debounceTime(300),
+  switchMap(
+    (searchInput) =>
+      searchInput.length ?
+      ajax.getJSON<ITickerSearchResult[]>(`/api/ticker-search/${searchInput}`).pipe(
+        map((searchResults) => {
+          return searchResults.map(
+            ({ symbol, sector }) =>
+              ({
+                label: `${symbol} - ${sector}`,
+                value: symbol,
+              } as TickerOption),
+          )
+        }) 
+      ) : of([] as TickerOption[])
+    // ToDo -> add catchError => propagate to ErrorBoundary
+  ))
+
+const [useOptions, options$] = bind(
+  mergeWithKey(
+    {
+      searchResults: searchResults$,
+      tickerSelections: tickerSelections$
+    }
+  ).pipe(
+    map(
+      (event) =>
+        event.type === 'searchResults' ? event.payload : [] as TickerOption[]
+    )
+  ),
+  [] as TickerOption[]
+)
+
+
+export { tickerSelections$ }
+
+function OptionsList({ options }: { options: TickerOption[] }) {
+  return (
+    <ul className={styles.dropdown}>
+      {options.map((option) => (
+        <li key={option.value}>
+          <button type="button" className={styles.dropdown__option} onClick={
+            () => {
+              onTickerSelection(option.value)
+            }
+            }>
+            {option.label}
+          </button>
+        </li>
+      ))}
+    </ul>
+  )
+}
+
+export default function TickerSearch() {
+  const options = useOptions()
+
+  return (
+    <form className={styles.search}>
```

Comment:
some functionality options:
1. the user is able to submit the form and send the input value as the stock ticker payload
2. the user is not able to submit the form, and must choose from the buttons in the drop down (so would need to re-add the preventDefault
3. the user is able to submit the form only if the input value matches one of the options
4. the user is able to submit the form and the first option is sent as the stock ticker payload

Reply:
Created new issue with suggestions